src/detect-byte.c

```diff
@@ -27,30 +27,63 @@
 #include "detect-byte-extract.h"
 #include "detect-bytemath.h"
 
+// Once a variable is found, apply strict semantics (error) else warning
+static bool DetectByteListMismatch(bool any, int sm_list, int found_list, bool strict,
+        const char *arg, const DetectEngineCtx *de_ctx)
+{
+    if (any || sm_list == found_list)
+        return false;
+
+    if (strict) {
+        return true;
+    }
+
+    if (de_ctx) {
```

Comment:
How can `de_ctx ` be NULL ?

Reply:
It'll be NULL when called from the Lua functions.